src/vulnpy/templates/fragments/nav.html

```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
```

Comment:
This is the HTML for the navigation components. Each `.frag.html` is injected into this file (statically) to produce a complete webpage. Modifications made to fragments must be followed by `make templates` to regenerate. The structure of html is:
```
vulnpy/templates
├── cmdi.html (generated)
├── home.html (generated)
├── fragments
    ├── cmdi.frag.html
    ├── home.frag.html
    └── nav.html
```
I know this is strange, so I'm really interested in what you guys think here.

Reply:
This is super interesting. This is definitely where using jinja2 and other frameworks help us via block injection using {{ }} and {% %} etc.

It's hard to take a look at the commonalities across each of the htmls, especially home html. versus cmdi.html. I think my only ask here is to make a base.html (which is a common pattern in templating frameworks) which houses all the base elements. I'd think either home or cmdi could become it? That way when we add xss.html it's obvious it has to "inherit" from base.html.

Reply:
I think I see what you're saying, and let me try to explain. First, `nav.html` is what you're probably thinking of as `base.html`, and I like your filename better, so I'm going to change it. However, with pure HTML (to the best of my limited understanding / furious googling) there's no way to import, include, or otherwise reuse code dynamically. I couldn't get it to work, and I tried for a really long time.

The solution I landed on eliminates code duplication for _non-generated_ files. So for development, making a change to `nav.html` (soon to be `base.html`) and running `make` is the only necessary change. If we need to make a new page, like `xss.html`, we just need to make `xss.frag.html`, which won't have any duplicated code, and rerun `make` which splices together `xss.frag.html` and `nav.html` to create `xss.html`.